enhancements/microshift/microshift-supportability-tools.md

```diff
@@ -0,0 +1,353 @@
+
+#### Variation [optional]
+N/A
+
+### API Extensions
+N/A
+
+### Implementation Details/Notes/Constraints [optional]
+N/A
+
+### Risks and Mitigations
+There are four main risks:
+1. Upgrades and incompatible changes.
+1. `sos` release cadence.
+1. CEE having a different tool.
+1. Exposing sensitive data.
+
+#### Upgrades
+When upgrading a MicroShift system we may need to upgrade just MicroShift, or
+the OS as well. If it is the latter then `sos` will be updated before
+MicroShift, therefore backwards compatibility with the plugin must be ensured.
+
+MicroShift will include `sos` as a dependency, tying them together in case of
+upgrades/downgrades.
+
+The only moment where there could be a mismatch is right after upgrading
+`sos`, and failing to upgrade `microshift`. In this case the gathering tool
+will be at a higher version than `microshift`. The opposite situation also
+holds true. Extra care must be taken in order to support these situations by
+using the minimum amount of hardcoded names and assumptions on the system. If
+there is a generic way of gathering information, it should be used.
+
+#### Sos release cadence
+`sos` release cadence has been shortened to 4 weeks starting on Feb 1st 2023.
+This change has been introduced to allow faster support of new components and
+features. Since MicroShift follows the same release calendar as OpenShift,
+`sos` is faster.
+
+New versions of `sos` will simply slot in to the next regular release of both
+RHEL 8 and 9. If there is an urgent need a hotfix can be made between releases.
+
+#### CEE having a different tool
+CEE might have concerns over using a different tool than `must-gather`. `sos`
+reports are able to extract a full system report, as opposed to `must-gather`
+which is only suitable for OpenShift clusters.
+
+Moreover, there is a growing number of plugins to support anything that runs
+on a RHEL system, making it a suitable go-to gathering tool when it comes to
+diagnosing system-wide issues.
+
+The microshift plugin for `sos` shall keep the same output format when it comes
+to cluster resources than `must-gather` in order to minimize differences and
+allow the same tooling that is currently used for OpenShift.
+
+#### Sensitive data
+Tool shall not dump contents from certificates and/or any other sensitive data.
+There is a `--clean` option which obfuscates hostnames, IP addresses, MAC
+addresses, and any text matching user defined input.
+
+### Drawbacks
+Discussed in risks and mitigations.
+
+## Design Details
+Refer to `sos` [documentation](https://github.com/sosreport/sos/wiki/How-to-Write-a-Plugin)
+for information on how to create plugins.
+
+The file structure should ressemble that of must-gather.
+
+### Open Questions [optional]
+1. Should `/var/lib/microshift` contents be added to the report? They contain
+   certificates and etcd data, mainly. This directory includes sensitive data.
+1. `etcd` plugin requires `etcdctl` to retrieve information. Do we need this
+   plugin?
```

Comment:
etcdctl is 23 MB on x86. I think we could afford to add it, but we would have to get it packaged properly. We should capture the need to do something about this in a separate ticket.